judge/ratings.py

```diff
@@ -120,7 +119,7 @@ def divconq(i, j):
             tanh_terms = []
             w_prev = 1.
             w_sum = 0.
-            for j, h in enumerate(historical_p[i]):
+            for j, h in enumerate([new_p[i]] + historical_p[i]):
```

Comment:
Interesting. If it's for this, it might make sense to use `itertools.chain([new_p[i]], historical_p[i])`.

Reply:
`chain` didn't have an impact on performance. Should I use `chain` anyway?

Reply:
Supposedly it's better as the lists grow larger, so yea.

Reply:
Benchmarking shows that `chain` is slower.
```
$ python3 -m timeit -s 'a=.1;b=[.2,.3,.4]*300' 'sum([a]+b)'
100000 loops, best of 3: 7.48 usec per loop
$ python3 -m timeit -s 'a=.1;b=[.2,.3,.4]*300;from itertools import chain' 'sum(chain([a],b))'
100000 loops, best of 3: 9.37 usec per loop
```

Reply:
I see, then just stick with concatenation.